### ISSUE TITLE
Improve report location of no-trailing-spaces by specifying end location

### ISSUE THREAD

**Issue by golopot** [eslint/eslint#12315]
**What rule do you want to change?**
no-trailing-spaces

**Does this change cause the rule to produce more or fewer warnings?**
No

**How will the change be implemented? (New option, new default behavior, etc.)?**
Change report location by adding end location.

**Please provide some example code that this change will affect:**

```js
/* eslint no-trailing-spaces: 2*/
// trailing spaces
a     
```

**What does the rule currently do for this code?**
Only reports the start location but not end location. Therefore my editor only display one  tilde instead of many tildes covering all trailing spaces.
 
**What will the rule do after it's changed?**
The report location will have both start location and end location.

**Are you willing to submit a pull request to implement this change?**
No

**Comment by mdjermanovic** [comment]
Hi @golopot , thanks for the issue!

I can confirm that the rule reports only the start location. It makes sense in this case to highlight the full range that would be automatically removed by the fixer.

I'm willing to champion the enhancement, though I'm not sure should this be rather accepted as a bug.
